### ISSUE TITLE
Programmatically zoom to coordinates

### ISSUE THREAD

**Issue by jimmyfishbean** [potree/potree#580]
I have a huge point cloud that covers a large area.  I load the point cloud and use the viewer.fitToScreen() method so the whole point cloud is loaded in the viewer.

I would like to be able to programmatically zoom to some given coordinates straight away after loading the point cloud, since it is no use to me to view the whole of the point cloud.  I would like to mimic the functionality of the double click (zoomToLocation()) but I do not have available the mouse event/position as this is always unknown.  The real world coordinates I would like to zoom to can also vary each time (depends on where the user wants to jump to in the Potree viewer upon start up).

Please could someone tell me if this is possible?  (For example, is the idea of having a huge point cloud that covers hundreds of miles squared sensible first of all - can Potree handle this?)

For this query/issue, I have deliberately kept the point cloud small in size, so I can provide screen shots.  The following screen shows the whole of my (example) point cloud, the range from the cloud.js being:

{
    "version": "1.7",
    "octreeDir": "data",
    "projection": "",
    "points": 964742813,
    "boundingBox": {
        "lx": 408696.327,
        "ly": 289601.47,
        "lz": 75.427,
        "ux": 415516.947,
        "uy": 296422.08999999999,
        "uz": 6896.046999999995
    },
    "tightBoundingBox": {
        "lx": 408696.327,
        "ly": 289601.47,
        "lz": 75.427,
        "ux": 415516.947,
        "uy": 291085.543,
        "uz": 158.661
    },
    "pointAttributes": "LAZ",
    "spacing": 59.068302154541019,
    "scale": 0.001,
    "hierarchyStepSize": 5
}

![image](https://user-images.githubusercontent.com/4130571/49372465-ad036700-f6f2-11e8-8fb2-bfe82ec5e49a.png)

I would like to zoom to the XYZ coordinates 409200.000, 290200.000, 150.000 which is roughly the top of the point cloud from the first screenshot.  The next screen shot shows the rough part of the point cloud I would like to load/zoom to:

![image](https://user-images.githubusercontent.com/4130571/49372664-4a5e9b00-f6f3-11e8-99e4-fc8527a1905e.png)

I have tried the following code and the point cloud is zoomed, but for some reason it does not zoom at the expected coordinates.  It seems to just zoom in at the center of the whole point cloud:

![image](https://user-images.githubusercontent.com/4130571/49372905-1c2d8b00-f6f4-11e8-8efc-b48b059299fd.png)

My code:

    Potree.loadPointCloud("pointclouds/OsgbColour/cloud.js", "OsgbColour", e => {
            let pointcloud = e.pointcloud;
            let material = pointcloud.material;
            viewer.scene.addPointCloud(pointcloud);
            material.pointColorType = Potree.PointColorType.RGB; // any Potree.PointColorType.XXXX 
            material.size = 1;
            material.pointSizeType = Potree.PointSizeType.ATTENUATED;
            material.shape = Potree.PointShape.CIRCLE;
            viewer.fitToScreen();

            viewer.scene.view.position.set(409200.000, 290200.000, 150.000);
            viewer.scene.view.lookAt(4092000.000, 290200.000, 150.000);
            var myactivecamera = viewer.scene.getActiveCamera();
            myactivecamera.position.set(4092000.000, 290200.000, 150.000);
            myactivecamera.zoom += 3.0;
        });

Thanks in advance.  

Jimmy

**Comment by jimmyfishbean** [comment]
I have managed so sort this using the following code (the last 2 lines):

    Potree.loadPointCloud("pointclouds/OsgbColour/cloud.js", "OsgbColour", e => {
            let pointcloud = e.pointcloud;
            let material = pointcloud.material;
            viewer.scene.addPointCloud(pointcloud);
            material.pointColorType = Potree.PointColorType.RGB; // any Potree.PointColorType.XXXX
            material.size = 1;
            material.pointSizeType = Potree.PointSizeType.ATTENUATED;
            material.shape = Potree.PointShape.CIRCLE;
            
            viewer.scene.view.position.set(409600.000, 290000.000, 400.000);
            viewer.scene.view.lookAt(new THREE.Vector3(409200.000, 290400.000, 75.000));
        });

**Comment by jimmyfishbean** [comment]
I closed this issue without commenting - as I have managed to sort it, as follows (the last 2 lines):

    Potree.loadPointCloud("pointclouds/OsgbColour/cloud.js", "OsgbColour", e => {
            let pointcloud = e.pointcloud;
            let material = pointcloud.material;
            viewer.scene.addPointCloud(pointcloud);
            material.pointColorType = Potree.PointColorType.RGB; // any Potree.PointColorType.XXXX
            material.size = 1;
            material.pointSizeType = Potree.PointSizeType.ATTENUATED;
            material.shape = Potree.PointShape.CIRCLE;
            
            viewer.scene.view.position.set(409600.000, 290000.000, 400.000);
            viewer.scene.view.lookAt(new THREE.Vector3(409200.000, 290400.000, 75.000));
        });

**Comment by EthanYi86** [comment]
Hi, Jimmy, I'm visualizing a large scale point cloud with potree now, almost the same size as yours. But when I zoomed in the point cloud for some details, the lane markers on the road for example, the point cloud was too sparseI. I measured the distances between two points, most of them is longer than 0.5m. Have you ever met similar problem? 

I run PotreeConverter with --spacing = 8, --level = 8, --scale = 0.01, the resolustion of the last level is supposed to be 0.03125m.

**Comment by jimmyfishbean** [comment]
Ethan.  Apologies but I have no experience at down scaling/reducing the number of points using the Potree Converter switches.  The tiles I used to convert had up to 20 million points in them - and the requirement was to provide as much detail as possible when rendering the point cloud (i,e, convert ALL points).  That is the main reason I switched to using Potree since it loads data on the fly (as you will know) so loading the full number of points was possible.

However, I would suggest running the Potree Converter to output ALL points initially and see if this fixes the issue.  If it does, then you will need to try and apply the switches.  I no longer work in the same company so cannot try this myself to assist as I have no access to LiDAR data.
